Paco/src/com/pacoapp/paco/model/ExperimentProviderUtil.java

```diff
@@ -869,12 +885,16 @@ private Event findEventBy(String select, String[] selectionArgs, String sortOrde
   private List<Output> findResponsesFor(Event event) {
```

Comment:
This is too granular a place for the lock. It should lock around all insertEvent and find/getEvent calls instead of at the level of the responses.

Any method that uses a contentResolver to get at EventColumns.CONTENT_URI needs to have the appropriate lock to keep the event table coherent.